CLI dist: ERR_MODULE_NOT_FOUND on extensionless imports (moduleResolution: bundler + Node ESM)

## Summary

After building with `npm run build`, running the compiled `ankiniki` binary fails with:

```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '.../apps/cli/dist/commands/note'
imported from .../apps/cli/dist/index.js
```

## Root cause

The root `tsconfig.json` uses `"moduleResolution": "bundler"`, which allows extensionless imports (e.g. `import { createNoteCommand } from './commands/note'`). Bundlers resolve these fine, but Node.js's native ESM loader requires explicit `.js` extensions. Since the CLI bin entry points directly to `dist/index.js` for Node to execute, it fails at runtime.

## Reproduction

```bash
npm install
npm run build
ankiniki config --show  # → ERR_MODULE_NOT_FOUND
```

## Workaround

Run via `tsx` directly:

```bash
npx tsx apps/cli/src/index.ts config --show
```

## Fix options

1. **`NodeNext` resolution** — change `moduleResolution` to `NodeNext` in the CLI's tsconfig and add `.js` extensions to all relative imports in `apps/cli/src`
2. **Bundle with esbuild** — keep `bundler` resolution but output a single bundled file; cleanest for a CLI tool, no resolution issues at runtime
3. **tsx launcher** — change the `bin` shebang to invoke `tsx` instead of `node`

Option 2 (esbuild bundle) is recommended for a CLI — single output file, zero resolution issues, fast startup.

--- a/apps/cli/build.mjs
+++ b/apps/cli/build.mjs
@@ -1,0 +1,21 @@
+import { build } from 'esbuild';
+import { chmodSync, mkdirSync } from 'fs';
+
+mkdirSync('dist', { recursive: true });
+
+await build({
+  entryPoints: ['src/index.ts'],
+  bundle: true,
+  platform: 'node',
+  format: 'cjs',
+  outfile: 'dist/index.js',
+  banner: {
+    js: '#!/usr/bin/env node',
+  },
+  // Node built-ins are automatically external with platform:node.
+  // anki-apkg-export uses a webpack script-loader shim that esbuild cannot
+  // resolve; it is a pre-built CJS package so Node loads it fine at runtime.
+  external: ['anki-apkg-export'],
+});
+
+chmodSync('dist/index.js', 0o755);

--- a/apps/cli/package.json
+++ b/apps/cli/package.json
@@ -6,7 +6,7 @@
     "ankiniki": "./dist/index.js"
   },
   "scripts": {
-    "build": "tsc",
+    "build": "node build.mjs",
     "dev": "tsx watch src/index.ts",
     "start": "node dist/index.js",
     "test": "vitest run",
@@ -28,6 +28,7 @@
     "@types/inquirer": "^9.0.3",
     "@types/node": "^20.5.0",
     "jest": "^29.6.2",
+    "esbuild": "^0.25.0",
     "tsx": "^3.12.7",
     "typescript": "^5.1.6",
     "vitest": "^4.1.2"

--- a/apps/cli/src/index.ts
+++ b/apps/cli/src/index.ts
@@ -1,5 +1,3 @@
-#!/usr/bin/env node
-
 import { Command } from 'commander';
 import chalk from 'chalk';
 import { createNoteCommand } from './commands/note';

